### ISSUE TITLE
Increasing validation loss during training

### ISSUE THREAD

**Issue by luca-ant** [tensorflow/recommenders#134]
I'm following the notebook of this tutorial ([deep_recommenders](https://www.tensorflow.org/recommenders/examples/deep_recommenders)) and I see that the dataset is split only in two parts (train and test). 

For other kind of tasks, like classification, I know that dataset is split in train, validation and test in order to use, for example, early stopping on validation metrics to avoid to overfit the model on train data. I think I can use early stopping also while I'm training deep neural networks for recommendations, but running the notebook mentioned before without any changes the loss on test set grows up during the training instead of decreasing. It is a normal behavior?

Another question (maybe related to this). Why is the batch size so large? (2048 for train set and 4096 for test set) I notice that changing it cause a big variation on loss value..

I report below some information about training params and metrics of last training epochs
> one_layer_history = model.fit(
    cached_train,
    validation_data=cached_test,
    validation_freq=5,
    epochs=300,
    verbose=1)

> Epoch 275/300
40/40 [==============================] - 9s 222ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11909.4069 - regularization_loss: 0.0000e+00 - total_loss: 11909.4069 - val_factorized_top_k/top_1_categorical_accuracy: 1.0000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0014 - val_factorized_top_k/top_10_categorical_accuracy: 0.0053 - val_factorized_top_k/top_50_categorical_accuracy: 0.1135 - val_factorized_top_k/top_100_categorical_accuracy: 0.2682 - val_loss: 32880.6445 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 32880.6445**
Epoch 276/300
40/40 [==============================] - 4s 92ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11908.4688 - regularization_loss: 0.0000e+00 - total_loss: 11908.4688
Epoch 277/300
40/40 [==============================] - 4s 93ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11905.8294 - regularization_loss: 0.0000e+00 - total_loss: 11905.8294
Epoch 278/300
40/40 [==============================] - 4s 92ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11906.1159 - regularization_loss: 0.0000e+00 - total_loss: 11906.1159
Epoch 279/300
40/40 [==============================] - 4s 90ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11906.5296 - regularization_loss: 0.0000e+00 - total_loss: 11906.5296
Epoch 280/300
40/40 [==============================] - 9s 215ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11906.5406 - regularization_loss: 0.0000e+00 - total_loss: 11906.5406 - val_factorized_top_k/top_1_categorical_accuracy: 2.5000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0019 - val_factorized_top_k/top_10_categorical_accuracy: 0.0052 - val_factorized_top_k/top_50_categorical_accuracy: 0.1126 - val_factorized_top_k/top_100_categorical_accuracy: 0.2677 - val_loss: 32925.2422 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 32925.2422**
Epoch 281/300
40/40 [==============================] - 3s 87ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11903.4211 - regularization_loss: 0.0000e+00 - total_loss: 11903.4211
Epoch 282/300
40/40 [==============================] - 3s 87ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11904.1313 - regularization_loss: 0.0000e+00 - total_loss: 11904.1313
Epoch 283/300
40/40 [==============================] - 4s 89ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11904.1110 - regularization_loss: 0.0000e+00 - total_loss: 11904.1110
Epoch 284/300
40/40 [==============================] - 4s 90ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11902.0310 - regularization_loss: 0.0000e+00 - total_loss: 11902.0310
Epoch 285/300
40/40 [==============================] - 9s 213ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11901.5600 - regularization_loss: 0.0000e+00 - total_loss: 11901.5600 - val_factorized_top_k/top_1_categorical_accuracy: 1.0000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0018 - val_factorized_top_k/top_10_categorical_accuracy: 0.0052 - val_factorized_top_k/top_50_categorical_accuracy: 0.1131 - val_factorized_top_k/top_100_categorical_accuracy: 0.2706 - val_loss: 32949.3906 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 32949.3906**
Epoch 286/300
40/40 [==============================] - 4s 88ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11901.8011 - regularization_loss: 0.0000e+00 - total_loss: 11901.8011
Epoch 287/300
40/40 [==============================] - 4s 88ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11900.6831 - regularization_loss: 0.0000e+00 - total_loss: 11900.6831
Epoch 288/300
40/40 [==============================] - 3s 87ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11900.3813 - regularization_loss: 0.0000e+00 - total_loss: 11900.3813
Epoch 289/300
40/40 [==============================] - 3s 87ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11900.7525 - regularization_loss: 0.0000e+00 - total_loss: 11900.7525
Epoch 290/300
40/40 [==============================] - 13s 322ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11899.1266 - regularization_loss: 0.0000e+00 - total_loss: 11899.1266 - val_factorized_top_k/top_1_categorical_accuracy: 1.0000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0018 - val_factorized_top_k/top_10_categorical_accuracy: 0.0049 - val_factorized_top_k/top_50_categorical_accuracy: 0.1132 - val_factorized_top_k/top_100_categorical_accuracy: 0.2705 - val_loss: 32987.5469 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 32987.5469**
Epoch 291/300
40/40 [==============================] - 3s 86ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11899.5831 - regularization_loss: 0.0000e+00 - total_loss: 11899.5831
Epoch 292/300
40/40 [==============================] - 3s 86ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11896.3421 - regularization_loss: 0.0000e+00 - total_loss: 11896.3421
Epoch 293/300
40/40 [==============================] - 3s 85ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11897.9094 - regularization_loss: 0.0000e+00 - total_loss: 11897.9094
Epoch 294/300
40/40 [==============================] - 3s 83ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11896.4027 - regularization_loss: 0.0000e+00 - total_loss: 11896.4027
Epoch 295/300
40/40 [==============================] - 8s 206ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11896.5980 - regularization_loss: 0.0000e+00 - total_loss: 11896.5980 - val_factorized_top_k/top_1_categorical_accuracy: 2.5000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0018 - val_factorized_top_k/top_10_categorical_accuracy: 0.0054 - val_factorized_top_k/top_50_categorical_accuracy: 0.1119 - val_factorized_top_k/top_100_categorical_accuracy: 0.2693 - val_loss: 33031.1914 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 33031.1914**
Epoch 296/300
40/40 [==============================] - 3s 83ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11896.8725 - regularization_loss: 0.0000e+00 - total_loss: 11896.8725
Epoch 297/300
40/40 [==============================] - 3s 84ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11894.4259 - regularization_loss: 0.0000e+00 - total_loss: 11894.4259
Epoch 298/300
40/40 [==============================] - 3s 84ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11894.5416 - regularization_loss: 0.0000e+00 - total_loss: 11894.5416
Epoch 299/300
40/40 [==============================] - 3s 84ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11893.9245 - regularization_loss: 0.0000e+00 - total_loss: 11893.9245
Epoch 300/300
40/40 [==============================] - 8s 203ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_5_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_10_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_50_categorical_accuracy: 0.0000e+00 - factorized_top_k/top_100_categorical_accuracy: 0.0000e+00 - loss: 11894.7852 - regularization_loss: 0.0000e+00 - total_loss: 11894.7852 - val_factorized_top_k/top_1_categorical_accuracy: 2.0000e-04 - val_factorized_top_k/top_5_categorical_accuracy: 0.0018 - val_factorized_top_k/top_10_categorical_accuracy: 0.0049 - val_factorized_top_k/top_50_categorical_accuracy: 0.1121 - val_factorized_top_k/top_100_categorical_accuracy: 0.2691 - val_loss: 33056.2227 - val_regularization_loss: 0.0000e+00 - **val_total_loss: 33056.2227**
Top-100 accuracy: 0.27.

**Comment by maciejkula** [comment]
Hey! One of the drawbacks of deep learning is that it requires very careful tuning, including things like learning rate and batch size. Our tutorials are intended as illustrations of different ways of building models, but will still require tuning.

In general, your intuition about validation loss decreasing is correct (but as always, [there are exceptions](https://openai.com/blog/deep-double-descent/)).

**Comment by luca-ant** [comment]
Thanks for clarification and for interesting reading! :nerd_face:
